regex-syntax/src/ast/parse.rs

```diff
@@ -2260,6 +2268,29 @@ impl<'p, 's, P: Borrow<Parser>> ast::Visitor for NestLimiter<'p, 's, P> {
     }
 }
 
+/// When the result is an error, transforms the ast::ErrorKind from the source
+/// Result into another one. This function is used to return clearer error
+/// messages when possible.
+fn specialize_err<T>(
+    result: Result<T>,
+    from: ast::ErrorKind,
+    to: ast::ErrorKind,
+) -> Result<T> {
+    if let Err(e) = result {
```

Comment:
This could be written with a match guard:

```rust
match result {
    Err(e) if e.kind == from => Err(ast::Error { kind: to, ..e }),
    _ => result,
}
```

Reply:
Wait nevermind this is moving